### PR TITLE
USER DMS

### PR DESCRIPTION
**Pull request by ankitpws**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: module/programUsers/helper.js [hunk 1]
```diff
@@ -5,160 +5,169 @@
...
-}
\ No newline at end of file
+  /**
+   * bulkUpdate function is used to do bulk operations.
+   * @method
+   * @name bulkUpdate
+   * @param {Object} [Data] -  Data along with filter and quries.
+   * @returns {Object} - status of bulk operation.
+   */
+  static bulkUpdate(Data) {
```
Comment (aks30) [inline module/programUsers/helper.js:161]: @ankitpws - This is not required

File: generics/services/users.js [hunk 2]
```diff
@@ -6,411 +6,430 @@
...
+            httpStatusCode["http_responsecode_ok"].message
+          ) {
+            result["data"] = response.result.response;
+            result.data = _.omit(response.result, [
+              "email",
+              "maskedEmail",
+              "maskedPhone",
+              "recoveryEmail",
+              "phone",
+              "lastName",
```
Comment (aks30) [inline generics/services/users.js:57]: @ankitpws From here and other services don't remove lastName

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 18 Nov. Do all changes as per project service comments as well

File: module/programUsers/helper.js (hunk 1 above)
Comment (ankitpws) [reply module/programUsers/helper.js]: removed this function instead of this we are using updateMany

File: generics/services/users.js (hunk 2 above)
Comment (ankitpws) [reply generics/services/users.js]: removed lastName from here

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 24 Nov 5 PM

File: config/kafka.js [hunk 3]
```diff
@@ -38,11 +39,55 @@ const connect = function() {
         console.log("kafka producer creation error!")
     })
 
+    _sendToKafkaConsumers(USER_DELETE_TOPIC, process.env.KAFKA_URL, USER_DELETE_ON_OFF);
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply config/kafka.js]: added

File: envVariables.js [hunk 4]
```diff
@@ -88,7 +88,27 @@ let enviromentVariables = {
...
+    message: "Required Service ID",
+    optional: false,
+  },
+  "TELEMETRY_ON_OFF":{
+    message: "Required telemetry on/off status",
+    optional: false,
+  },
+  "TELEMETRY_TOPIC": {
+    message: "Required telemetry topic",
+    optional: true,
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply envVariables.js:117]: used

File: release-notes/7.0.0.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+### New Environment Keys Added
+
+We added new environment keys to the DevOps repository ([PR link](https://github.com/project-sunbird/sunbird-devops/pull/3921/files)) to as required for new features and functionality. For configuration and access to outside services or resources, these keys will be utilised.
+
+Please note you don't need to deploy the DevOps repo. Once the PR is merged, deploy this service, env variable will automatically add from the DevOps branch.
+
+In this release, we have introduced four new environment variables.
+
+    USER_DELETE_TOPIC={{user_delete_topic_name | default("")}}
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply release-notes/7.0.0.md]: defined

File: config/kafka.js [hunk 6]
```diff
@@ -8,7 +8,8 @@
 
 //dependencies
 const kafka = require('kafka-node');
-
+let USER_DELETE_TOPIC = process.env.USER_DELETE_TOPIC;
+let USER_DELETE_ON_OFF = process.env.USER_DELETE_ON_OFF
```
[the remark this answers is not in the document]
Comment (ankitpws) [reply config/kafka.js]: used

**Pull request merged**